vault.go

```diff
@@ -0,0 +1,320 @@
+package chef
+
+import (
+	"crypto/sha256"
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"strings"
+)
+
+const algorithm string = "aes-256-gcm"
+const supportedVersion int = 3
+
+var errUnsupportedVersion = errors.New("Only Version 3 encrypted values are supported")
+
+// VaultService is the service for interacting with the chef server data endpoint
+type VaultService struct {
+	client *Client
+}
+
+// VaultItem wraps a vault databag and it's keys
+type VaultItem struct {
+	DataBagItem  *DataBagItem
+	Keys         *VaultItemKeys
+	Name         string
+	Vault        string
+	VaultService *VaultService
+}
+
+// VaultItemKeys contains the client keys associated with a
+type VaultItemKeys struct {
+	DataBagItem *DataBagItem
+	Name        string
+}
+
+// VaultListResult is the list of vaults returned by chef-api when listing
+// http://docs.getchef.com/api_chef_server.html#data
+type VaultListResult map[string]string
+
+// String makes VaultListResult implement the string result
+func (vlr VaultListResult) String() (out string) {
+	for k, v := range vlr {
+		out += fmt.Sprintf("%s => %s\n", k, v)
+	}
+	return out
+}
+
+// List returns a list of vaults on the server.  The official implementation of this method loads each databag and makes some assumptions based on the number of keys and their names.
+// See https://sourcegraph.com/github.com/chef/chef-vault@0d008c3/-/blob/lib/chef/knife/vault_list.rb#L38-51
+func (vs *VaultService) List() (*VaultListResult, error) {
+	databags, err := vs.client.DataBags.List()
+	if err != nil {
+		return nil, err
+	}
+
+	keys := map[string]bool{}
+
+	for name := range *databags {
+		if vaultName := strings.TrimSuffix(name, "_keys"); vaultName != name {
```

Comment:
I don't believe this code finds chef vault data bags.  The vaults don't have the _keys name, the items within the vaults have those names.